### PR TITLE
fix #573 encoding of waytype ferry

### LINKED ISSUE #573: waytype=ferry does not resolve for extra_info directions request
If ferries are part of the itinerary, the parameter [`extra_info=[waytype]`](https://github.com/GIScience/openrouteservice-docs#waytype) does not resolve the ferries (ID 9) and instead responds with ID 0 (unknown) to the ferry sections. 

E.g. Konstanz - Friedrichshafen:
```
curl -X POST \
  'http://api.openrouteservice.org/v2/directions/driving-hgv/json' \
  -H 'Content-Type: application/json' \
  -d '{
    "coordinates": [
        [
            9.169464,
            47.659607
        ],
        [
            9.493561,
            47.666312
        ]
    ],
    "extra_info": [
        "waycategory",
        "waytype"
    ],
    "instructions": false,
    "geometry": false
}'
```

### PR DESCRIPTION
**Pull request by takb**
### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the development branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [x] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes #573.

### Information about the changes
- Reason for change:
WaySurfaceTypeGraphStorageBuilder class was not encoding the WayType for ferry routes that did not also have a highway tag.

### CODE REVIEW

**Review by rabidllama — APPROVED** [review]

File: openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java [hunk 1]
```diff
@@ -54,37 +52,37 @@ public void processWay(ReaderWay way) {
 		boolean hasHighway = way.hasTag("highway");
 		boolean isFerryRoute = way.hasTag("route", ferries);
 
-		java.util.Iterator<Entry<String, Object>> it = way.getProperties();
+		Iterator<Entry<String, Object>> it = way.getProperties();
+
+		if (isFerryRoute)
+            waySurfaceDesc.WayType = ((byte)WayType.Ferry);
```
Comment (sfendrich) [inline openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java]: I'd prefer to always use braces around the if-body for readability.

File: openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java [hunk 2]
```diff
@@ -54,37 +52,37 @@ public void processWay(ReaderWay way) {
...
-							waySurfaceDesc.SurfaceType = (byte)SurfaceType.Unpaved;
-					}
+                    byte wayType = (isFerryRoute) ? WayType.Ferry : (byte) WayType.getFromString(value);
 
-					waySurfaceDesc.WayType = wayType;
-				}
-				else if (key.equals("surface")) {
+                    if (waySurfaceDesc.SurfaceType == 0) {
+                        if (wayType == WayType.Road || wayType == WayType.StateRoad || wayType == WayType.Street)
+                            waySurfaceDesc.SurfaceType = (byte) SurfaceType.Asphalt;
```
Comment (sfendrich) [inline openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java]: Braces again.

File: openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java [hunk 3]
```diff
@@ -54,37 +52,37 @@ public void processWay(ReaderWay way) {
...
+                        else if (wayType == WayType.Path)
+                            waySurfaceDesc.SurfaceType = (byte) SurfaceType.Unpaved;
```
Comment (sfendrich) [inline openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java]: Braces again.

File: openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java [hunk 4]
```diff
@@ -54,37 +52,37 @@ public void processWay(ReaderWay way) {
...
+                    }
+                    if (waySurfaceDesc.WayType == 0)
+                        waySurfaceDesc.WayType = wayType;
```
Comment (sfendrich) [inline openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java]: Braces again.

**Review by sfendrich — CHANGES_REQUESTED** [review]
Minor cleanup required (see inline comments). Also the indentation seems a bit messy.

File: openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java (hunk 4 above)
Comment (takb) [reply openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/storages/builders/WaySurfaceTypeGraphStorageBuilder.java]: done

**Review by sfendrich — APPROVED** [review]

**Pull request merged**